xbmc/music/tags/TagLoaderTagLib.cpp

```diff
@@ -1048,10 +1057,27 @@ void CTagLoaderTagLib::SetGenre(CMusicInfoTag &tag, const std::vector<std::strin
 
 void CTagLoaderTagLib::SetReleaseType(CMusicInfoTag &tag, const std::vector<std::string> &values)
 {
+  tag.SetBoxset(false);
   if (values.size() == 1)
+  {
+    if (values[0].find("boxset") != std::string::npos)
+    {
+      tag.SetBoxset(true);
+      if (tag.GetDiscSubtitle().empty()) // no disc title for the tracks on this boxset disc
+        tag.SetDiscSubtitle("dummy");
+    }
     tag.SetMusicBrainzReleaseType(values[0]);
+  }
   else
+  {
     tag.SetMusicBrainzReleaseType(StringUtils::Join(values, CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_musicItemSeparator));
+    if (tag.GetMusicBrainzReleaseType().find("boxset") != std::string::npos)
+    {
+      tag.SetBoxset(true);
+      if (tag.GetDiscSubtitle().empty())
+        tag.SetDiscSubtitle("dummy");
+    }
+  }
 }
```

Comment:
Ah well, the historical reason for this is that boxsets are **required** to have a title and that as it is possible to force any album to be a boxset via the tag I needed to generate a dummy title for any disc in such a set that doesn't have one set in the tags.

Looking back at what I've written, perhaps the place to do this (without having to alter all the logic of auto-detecting boxsets with titles) would be in `CMusicInfoScanner::CheckBoxSets`.  If the tag is set and the title is empty then set the dummy value there.

Hmmm (mini-groan right there !!), I'll re-visit it then and see if I can improve things somewhat.

Reply:
Thanks. Sorry for commenting on this so late in the day, I think I picked it up earlier then lost sight of it. Processing of tag values (finding "boxset" in album type, checking disc subtiles etc.) is definitely a `CMusicInfoScanner` thing. 

Overall I'm very pleased with this nice new feature, I think users will be too. Good job!

Reply:
Thanks!!  OK, so I've reworked that (see latest commit) and added a better comment to explain that basically I'm using an empty existing variable as a flag to set the dummy titles for the db with.

Now I have a question!  Actually, I noticed this a while ago, but forgot about it until I was back into tag reading again today.  At least two nodes (boxsets & compilations) don't appear until _all_ the scanning is done.  Means if you do an online scrape after the tag reading, it can take a while for those nodes to actually appear.  So, is there any reason I can't update the library bools for those in `CMusicDatabase::CommitTransaction()` or would that provide too much load ?  Can't say I'm bothered about the compilation node, just the boxsets one.